src/main/scala/cromwell/binding/expression/WdlStandardLibraryFunctions.scala

```diff
@@ -9,25 +9,78 @@ import scala.util.{Failure, Success, Try}
 trait WdlStandardLibraryFunctions extends WdlFunctions[WdlValue] {
   private def fail(name: String) = Failure(new UnsupportedOperationException(s"$name() not implemented yet"))
 
+  private def extractObjectOrArray(params: Seq[Try[WdlValue]]) = for {
+    singleArgument <- extractSingleArgument(params)
+    contents <- Success(fileContentsToString(singleArgument))
+    wdlObjects <- WdlObject.fromTsv(contents)
+  } yield wdlObjects
+
   protected def stdout(params: Seq[Try[WdlValue]]): Try[WdlFile] = fail("stdout")
   protected def stderr(params: Seq[Try[WdlValue]]): Try[WdlFile] = fail("stderr")
-  protected def read_lines(params: Seq[Try[WdlValue]]): Try[WdlArray] = fail("read_lines")
-  protected def read_tsv(params: Seq[Try[WdlValue]]): Try[WdlArray] = fail("read_tsv")
-  protected def read_map(params: Seq[Try[WdlValue]]): Try[WdlMap] = fail("read_map")
-  protected def read_object(params: Seq[Try[WdlValue]]): Try[WdlObject] = fail("read_objects")
-  protected def read_objects(params: Seq[Try[WdlValue]]): Try[WdlArray] = fail("read_objects")
+
+  /**
+    * Read all lines from the file referenced by the first parameter and return an Array[String]
+    */
+  protected def read_lines(params: Seq[Try[WdlValue]]): Try[WdlArray] = {
+    for {
+      singleArgument <- extractSingleArgument(params)
+      lines = fileContentsToString(singleArgument).split("\n").map{WdlString}
+    } yield WdlArray(WdlArrayType(WdlStringType), lines)
+  }
+
+ protected def read_map(params: Seq[Try[WdlValue]]): Try[WdlMap] = {
+    for {
+      singleArgument <- extractSingleArgument(params)
+      contents <- Success(fileContentsToString(singleArgument))
```

Comment:
Hmm not sure what tests or expected behaviors might be broken by changing that behavior.  We could just have one base method that reads an untrimmed string and then everything including a trimming `read_string` could be implemented in terms of that.